docs/docs/dart_api/dataclass.md

```diff
@@ -1,23 +1,112 @@
 ---
 
-title: Custom row classes
-description: Use your own classes as data classes for drift tables
+title: Dataclass
+description: Dataclass for reading and writing data to the database.
 
 ---
 
-For each table declared in Dart or in a drift file, `drift_dev` generates a row class (sometimes also referred to as _data class_)
-to hold a full row and a companion class for updates and inserts.
-This works well for most cases: Drift knows  what columns your table has, and it can generate a simple class for all of that.
-In some cases, you might want to customize the generated classes though.
-For instance, you might want to add a mixin, let it extend another class or interface, or use other builders like
-`json_serializable` to customize how it gets serialized to json.
 
-As a solution, drift allows you to use your own classes as data classes for the database.
 
-## Using custom classes
+# Generated Dataclass
 
-To use a custom row class, simply annotate your table definition with `@UseRowClass`.
+Drift generates a dataclass for each table in your database. These dataclasses represent query results and come with built-in equality, hashing, and serialization support. They also include a `copyWith` method for easy modification.
+
+**Example:**
+
+For a `Users` table, Drift automatically generates a `User` dataclass. This dataclass is used for all read operations from the `Users` table, ensuring type-safe and structured data retrieval.
+
+{{ load_snippet('generated-dataclass','lib/snippets/dart_api/dataclass.dart.excerpt.json') }}
+
+## Dataclass Name
+
+The dataclass name is derived from the table name.
+
+- If the name ends in `s`, the dataclass name will be the name with `s` removed.
+    - Example: `Users` -> `User`
+- Otherwise, the dataclass name will be the name with `Data` appended.
+    - Example: `UserInfo` -> `UserInfoData`
+
+
+To use a custom name use the `@DataClassName` annotation.
+
+**Example:**
+
+{{ load_snippet('data-class-name','lib/snippets/dart_api/dataclass.dart.excerpt.json') }}
+
+## Json serialization
```

Comment:
I think this should not be this far up  - and there should be a little warning saying that using custom row classes is the preferred approach for JSON serialization.

Reply:
Why?

Reply:
Drift is a relational persistence library, and ideally shouldn't have to do anything to do with serialization. It's a historical artifact that I regret, but it's too late to fix this now.

Our serialization capabilities are incredibly restricted compared to packages that actually intend to do serialziation, which is why I recommend using custom row classes to combine drift with other libraries where necessary.